src/compiler/pre-transform/codemods/legacy-story.test.ts

```diff
@@ -0,0 +1,195 @@
+      `
+      "<Story name="With source as text" parameters={{
+      	docs: {
+      		source: { code: "<LegacyStory>Hi</LegacyStory>" }
+      	}
+      }}>
+      	<LegacyStory>{'Hi'}</LegacyStory>
+      </Story>"
+    `
+    );
+  });
+
+  it("'source' prop when is a text expression gets moved to existing 'parameters'", async ({
+    expect,
+  }) => {
+    const code = `
+      <script context="module">
+        import { Story } from "@storybook/addon-svelte-csf";
+      </script>
+
+      <Story
+        name="Default"
+        source="'<Button primary />'"
+        parameters={{
+          controls: { disable: true },
+          interactions: { disable: true },
+        }}
+      />
+    `;
+    const node = await parseAndExtractSvelteNode<Component>(code, 'Component');
+
+    expect(print(transformLegacyStory({ node }))).toMatchInlineSnapshot(
+      `
+			"<Story name="Default" parameters={{
+				controls: { disable: true },
+				interactions: { disable: true },
+				docs: { source: { code: "'<Button primary />'" } }
+			}} />"
+		`
+    );
+  });
+
+  it("transforms 'template' prop to 'children' and text expression becomes expression tag with identifier to snippet", async ({
+    expect,
+  }) => {
+    const code = `
+      <script context="module">
+        import { Story } from "@storybook/addon-svelte-csf";
+      </script>
+
+      <Story name="Default" template="someTemplate" />
+    `;
+    const node = await parseAndExtractSvelteNode<Component>(code, 'Component');
+
+    expect(print(transformLegacyStory({ node }))).toMatchInlineSnapshot(
+      `"<Story name="Default" children={someTemplate} />"`
+    );
+  });
+
+  it("when directive 'let:args' is used then it wraps Story fragment with 'children' snippet block", async ({
+    expect,
+  }) => {
+    const code = `
+      <script context="module">
+        import { Story } from "@storybook/addon-svelte-csf";
+      </script>
+
+      <Story name="Default" let:args>
+        <Button {...args} />
+      </Story>
+    `;
+    const node = await parseAndExtractSvelteNode<Component>(code, 'Component');
+
+    expect(print(transformLegacyStory({ node }))).toMatchInlineSnapshot(`
+			"<Story name="Default">
+				{#snippet children(args)}
+					<Button {...args} />
+				{/snippet}
+			</Story>"
+		`);
+  });
+
+  it("when directive 'let:context' is used then it wraps Story fragment with 'children' snippet block", async ({
+    expect,
+  }) => {
+    const code = `
+      <script context="module">
+        import { Story } from "@storybook/addon-svelte-csf";
+      </script>
+
+      <Story name="Default" let:context>
+        <p>{context.id}</p>
+      </Story>
+    `;
+    const node = await parseAndExtractSvelteNode<Component>(code, 'Component');
+
+    expect(print(transformLegacyStory({ node }))).toMatchInlineSnapshot(`
+			"<Story name="Default">
+				{#snippet children(_args, context)}
+					<p>{context.id}</p>
+				{/snippet}
+			</Story>"
+		`);
+  });
+});
```

Comment:
can we add a test more that uses both `args` and `context` directives?

Reply:
2468f04cc9f9